### PR TITLE
[now-static-build] Add VuePress optimization

### PR DESCRIPTION
**Pull request by kidonng**
Origin: https://github.com/zeit/now-builders/pull/945
Test fixtures have been changed to VuePress's docs (with modifications).

### CODE REVIEW

**Comment by kidonng** [comment]
@styfle @AndyBitz I figured out why the test failed eventually.

https://github.com/zeit/now/blob/f6f99fef2529d26ab96ea4f525167d5873ec5db7/test/lib/deployment/test-deployment.js#L32

This won't match VuePress's `.vuepress` folder, so the deployment source files don't contain it and VuePress will use default config:

My own deployment:
image

Test deployment:

image

Seems we need to modify test script (in order to match `.vuepress`) to pass the test.

**Comment by AndyBitz** [comment]
@kidonng can you change the pattern to `/**/*` and see if the test pass then?

> This won't match VuePress's .vuepress folder, so the deployment source files don't contain it and VuePress will use default config

What exactly does "default config" mean here? Will the site still build, but with different content?

**Review by styfle — COMMENTED** [review]
https://github.com/isaacs/node-glob#dots

You can try adding `{dot: true}` to the glob options.

File: packages/now-static-build/src/frameworks.ts [hunk 1]
```diff
@@ -262,4 +262,23 @@ export default [
       },
     ],
   },
+  {
+    name: 'VuePress',
+    dependency: 'vuepress',
+    getOutputDirName: async () => 'docs/.vuepress/dist',
```
Comment (styfle) [inline packages/now-static-build/src/frameworks.ts:268]:
```suggestion
    getOutputDirName: async () => '.vuepress/dist',
```

The documentation says the default output directory is `.vuepress/dist`.

https://v1.vuepress.vuejs.org/guide/getting-started.html#inside-an-existing-project

**Comment by leo** [comment]
We just cleaned up the repository by deleting all our branches and re-creating only one of them. This now allows us to create canary and stable releases without having to deal with two separate branches.

As a result, this PR was automatically closed (not manually, even though the UI indicates that).

The idea looks like a great one! If you're interested, I'd love to welcome your PR to the codebase!

**Pull request closed without merging**